ui/app/components/app/metamask-translation/metamask-translation.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import MetaMaskTemplateRenderer, {
+  SectionShape,
+} from '../metamask-template-renderer/metamask-template-renderer';
+
+/**
+ * MetaMaskTranslation is a simple helper Component for adding full translation
+ * support to the template system. We do pass the translation function to the
+ * template getValues function, but passing it react components as variables
+ * would require react to be in scope, and breaks the object pattern paradigm.
+ *
+ * This component gets around that by converting variables that are templates
+ * themselves into tiny React trees. This component does additional validation
+ * to make sure that the tree has a single root node, with maximum two leaves.
+ * Each subnode can have a maximum of one child that must be a string.
+ *
+ * This enforces a maximum recursion depth of 2, preventing translation strings
```

Comment:
We might already have a case that doesn't meet these criteria: a link with a tooltip. The tooltip wrapper would be the first component, then the link would be the child, then the link text would be the child of the link.

Maybe the template components won't work like that, I don't know. I guess we'll see, and we can always adjust this later if we need to.

We'll need to be careful not to let this become a perverse incentive, and design weird components like LinkWithTooltip to get around this.